hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/TokenBalanceRepository.java

```diff
@@ -40,6 +40,30 @@ insert into token_balance (account_id, balance, consensus_timestamp, token_id)
     @Transactional
     int balanceSnapshot(long consensusTimestamp);
 
+    @Modifying
+    @Query(
+            nativeQuery = true,
+            value =
+                    """
+        with token_balance_snapshot as (
+          select account_id, balance, token_id from token_balance
+          where (account_id, token_id, consensus_timestamp)
+            in (select account_id, token_id, max(consensus_timestamp)
+                from token_balance
+                where consensus_timestamp >= :lowerRangeTimestamp and consensus_timestamp < :upperRangeTimestamp
+                group by account_id, token_id)
+        )
+        insert into token_balance (account_id, balance, consensus_timestamp, token_id)
+        select ta.account_id, ta.balance, :consensusTimestamp, ta.token_id
+        from token_account ta left join token_balance_snapshot tbs on ta.account_id = tbs.account_id and
+          ta.token_id = tbs.token_id
+        where ta.associated is true and
+          (tbs is null or tbs.balance is null or ta.balance <> tbs.balance)
+        order by ta.account_id, ta.token_id\s
```

Comment:
interesting this didn't cause any issue

```suggestion
        order by ta.account_id, ta.token_id
```

Reply:
Thanks I have removed that typo.